src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java

```diff
@@ -913,11 +913,11 @@ public void updateLocation(long userId, double latitude, double longitude) {
 
     // only update if latitude and longitude are not 0
     // else keep them at default 0, i.e. do nothing
-    if (latitude != 0 && longitude != 0) {
-      user.setLatitude(latitude);
-      user.setLongitude(longitude);
+    if (latitude == 0 && longitude == 0) {
+        return;
     }
-
+    user.setLatitude(latitude);
+    user.setLongitude(longitude);
```

Comment:
Why did you change that though?

Reply:
Because you could be on the equator or the greenwich line and then you would want it to update to that location. 0,0 is highly unlikely that it is ever reached so this is used to differenciate between a valid and invalid change of the location.

Reply:
Ah yes, stupid me. But you could just replaced && with || ^^